app/grandchallenge/components/views.py

```diff
@@ -472,3 +499,132 @@ def get(self, request, *args, **kwargs):
             },
         )
         return HttpResponse(html_content)
+
+
+class FileWidgetSelectView(LoginRequiredMixin, View):
+    def get(self, request, *args, **kwargs):
+        prefixed_interface_slug = self.request.GET.get(
+            "prefixed-interface-slug"
+        )
+        interface = get_object_or_404(
+            ComponentInterface,
+            slug=prefixed_interface_slug.replace(
+                INTERFACE_FORM_FIELD_PREFIX, ""
+            ),
+        )
+        widget_choice_name = request.GET.get(
+            f"widget-choice-{prefixed_interface_slug}"
+        )
+        try:
+            widget_choice = FileWidgetChoices(widget_choice_name)
+        except ValueError:
+            raise Http404(f"Widget choice {widget_choice_name} not found")
+        help_text = request.GET.get("help-text")
+        current_value = request.GET.get("current-value")
+
+        match widget_choice:
+            case FileWidgetChoices.FILE_SEARCH:

```

Comment:
Style wise I think we should only use `match`-`case` if an `if`-`elif` will do, which in this case it is as it is only doing a simple comparison each time. The rebinding problem with `match`-`case` is too easy to trigger and can cause serious bugs.